aptos-move/block-executor/src/scheduler.rs

```diff
@@ -600,8 +601,53 @@ impl Scheduler {
                 self.halt_transaction_execution(txn_idx);
             }
         }
+    }
+
+    /// Currently, the reasons for halting the scheduler are as follows:
+    /// 1. There is a module publishing txn that has read/write intersection with any txns
```

Comment:
Now when we do have a mutex protected status, we can actually have all of these be a separate status.

Reply:
I think this one is useful - to understand and verify code thats happening here